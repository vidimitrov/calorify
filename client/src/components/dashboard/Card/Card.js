import styled from 'styled-components';

const Card = styled.div`
  background-color: #fefbf0;
  font-family: Roboto;
  color: #f6cd48;
  width: 98%;
  height: 100px;
  display: flex;
  flex-direction: row;
  align-items: center;
  margin: 0 auto 20px;
  padding: 10px;
  border-radius: 5px;
  box-sizing: border-box;
  box-shadow: 0px 2px 4px -1px rgba(254, 244, 192, 0.2), 
              0px 4px 5px 0px rgba(0, 0, 0, 0.08), 
              0px 1px 10px 0px rgba(0, 0, 0, 0.05);
  
  @media (max-width: 420px) {
    height: 150px;
  }
`;

export default Card;
