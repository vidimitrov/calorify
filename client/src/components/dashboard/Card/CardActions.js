import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import IconButton from '@material-ui/core/IconButton';
import Edit from '@material-ui/icons/Edit';
import Delete from '@material-ui/icons/Delete';

const Wrapper = styled.div`
  width: 100px;

  ${props => props.wide && css`
    width: 150px;
  `}

  @media (max-width: 420px) {
    width: 80px;
  }
`;

function CardActions({
  onEditHandler,
  onDeleteHandler,
  customAction,
}) {
  return (
    <Wrapper wide={!!customAction}>
      {customAction
        && (
          <IconButton
            name={customAction.name ? customAction.name : ''}
            title={customAction.title ? customAction.title : ''}
            onClick={customAction.handler}
            color="default"
          >
            {customAction.icon}
          </IconButton>
        )
      }
      <IconButton
        onClick={onEditHandler}
        name="edit-meal"
        color="default"
      >
        <Edit />
      </IconButton>
      <IconButton
        onClick={onDeleteHandler}
        name="delete-meal"
        color="default"
      >
        <Delete />
      </IconButton>
    </Wrapper>
  );
}

CardActions.propTypes = {
  onEditHandler: PropTypes.func.isRequired,
  onDeleteHandler: PropTypes.func.isRequired,
  customAction: PropTypes.shape({
    icon: PropTypes.element.isRequired,
    handler: PropTypes.func.isRequired,
    title: PropTypes.string,
  }),
};

CardActions.defaultProps = {
  customAction: null,
};

export default CardActions;
