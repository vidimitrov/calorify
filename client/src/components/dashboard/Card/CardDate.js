import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import moment from 'moment';

const Wrapper = styled.div`
  color: #f0cd61;
  margin: 0px 20px;
  max-width: 100px;
`;

const FormattedDate = styled.h3`
  margin: 0;
  color: #c7b26e;

  @media (max-width: 420px) {
    font-size: 18px;
  }
`;

const CardDate = ({ date }) => {
  const formattedDate = moment(date).calendar();

  return (
    <Wrapper>
      <FormattedDate>{formattedDate}</FormattedDate>
    </Wrapper>
  );
};

CardDate.propTypes = {
  date: PropTypes.string.isRequired,
};

export default CardDate;
