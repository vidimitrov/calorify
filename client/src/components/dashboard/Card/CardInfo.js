import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  color: #f0cd61;
  margin: 0px 20px;
  flex-grow:1;

  @media (max-width: 420px) {
    min-width: 105px;
  }
`;

const Text = styled.h3`
  margin: 0;
  color: #c7b26e;

  @media (max-width: 420px) {
    font-size: 16px;
  }
`;

const Calories = styled.h1`
  margin: 0;

  @media (max-width: 420px) {
    font-size: 18px;
  }
`;

const CardInfo = ({ text, calories }) => (
  <Wrapper>
    <Calories>
      {calories}
      {' '}
      kCal
    </Calories>
    <Text>{text}</Text>
  </Wrapper>
);

CardInfo.propTypes = {
  text: PropTypes.string.isRequired,
  calories: PropTypes.number.isRequired,
};

export default CardInfo;
