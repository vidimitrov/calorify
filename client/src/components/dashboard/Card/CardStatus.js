import styled from 'styled-components';

const CardStatus = styled.div`
  background-color: ${props => (props.inRange ? 'rgb(153, 204, 0)' : '#ff387b')};
  height: 100%;
  width: 10px;
  border-radius: 3px;
`;

export default CardStatus;
