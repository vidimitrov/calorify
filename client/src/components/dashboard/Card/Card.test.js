import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Card from './Card';

it('Dashboard/Card/Card component renders correctly', () => {
  const tree = renderer.create(<Card />).toJSON();
  expect(tree).toMatchSnapshot();
});
