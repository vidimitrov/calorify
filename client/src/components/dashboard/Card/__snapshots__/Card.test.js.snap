// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard/Card/Card component renders correctly 1`] = `
.c0 {
  background-color: #fefbf0;
  font-family: Roboto;
  color: #f6cd48;
  width: 98%;
  height: 100px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 auto 20px;
  padding: 10px;
  border-radius: 5px;
  box-sizing: border-box;
  box-shadow: 0px 2px 4px -1px rgba(254,244,192,0.2),0px 4px 5px 0px rgba(0,0,0,0.08),0px 1px 10px 0px rgba(0,0,0,0.05);
}

@media (max-width:420px) {
  .c0 {
    height: 150px;
  }
}

<div
  className="c0"
/>
`;
